### ISSUE TITLE
Redacted reaction appears as latest message in room list

### ISSUE THREAD

**Issue by kittykat** [element-hq/element-android#6532]
### Steps to reproduce

1. In a room, react to a message
2. Check room list to see that reaction is visible as the latest message in the room
3. Remove the reaction
4. Check room list again

#### What did you expect?
Removed reaction is not shown as the latest message in the room when looking at the room list

#### What happened instead?
Removed reaction persists as the most recent message in the room when you look at the room list

### Operating system version

Android 11

### Application version and app store

1.4.27

### Homeserver

matrix.org

### Will you send logs?

No

### Are you willing to provide a PR?

No

**Comment by bmarty** [comment]
I have a quick fix but in this case, this is what will be displayed on the room list:

<img width="338" alt="image" src="https://user-images.githubusercontent.com/3940906/178823153-68777ce4-b69d-40a0-8a1c-0ef37836f587.png">

This is maybe not what is expected. I guess we want to see the latest unredacted event.

**Comment by kittykat** [comment]
Yes, as a user, I would prefer to see the last unredacted event but what you've got above is at least a suitable quickfix for now.
